core/drivers/gic.c

```diff
@@ -395,16 +395,18 @@ static void gic_it_raise_sgi(struct gic_data *gd __maybe_unused, size_t it,
 #if defined(CFG_ARM_GICV3)
 	/* Only support sending SGI to the cores in the same cluster now */
 	uint32_t mask_id = it & 0xf;
-	uint32_t mask_cpu = cpu_mask & 0xff;
+	uint32_t target_list = cpu_mask & 0xff;
 	uint64_t mpidr = read_mpidr();
 	uint64_t mask_aff1 = (mpidr & MPIDR_AFF1_MASK) >> MPIDR_AFF1_SHIFT;
 	uint64_t mask_aff2 = (mpidr & MPIDR_AFF2_MASK) >> MPIDR_AFF2_SHIFT;
 	uint64_t mask_aff3 = (mpidr & MPIDR_AFF3_MASK) >> MPIDR_AFF3_SHIFT;
-	uint64_t mask = (mask_cpu |
-			SHIFT_U64(mask_aff1, 16) |
-			SHIFT_U64(mask_id, 24)   |
-			SHIFT_U64(mask_aff2, 32) |
-			SHIFT_U64(mask_aff3, 48));
+	uint64_t mask = SHIFT_U64(mask_aff1, 16) | SHIFT_U64(mask_id, 24) |
+			SHIFT_U64(mask_aff2, 32) | SHIFT_U64(mask_aff3, 48);
+
+	if (cpu_mask & ITR_CPU_MASK_TO_THIS_CPU)
+		target_list = BIT32(mpidr & 0xf);
```

Comment:
Shouldn't the whole bitfield of affinity 0 be covered? I see no limitations on the number of CPU per cluster

target_list = BIT32(mpidr & 0xff)?

More than 4 core per cluster is quite unusual though.